src/provider/model/FinEnvironment.ts

```diff
@@ -61,6 +61,15 @@ export class FinEnvironment extends AsyncInit implements Environment {
     private _windowsCreated: number = 0;
     private readonly _seenWindows: {[id: string]: SeenWindow} = {};
 
+    public async isRunning(appInfo: Application): Promise<boolean> {
+        // TODO: This can be simplified once we change how we map between OpenFin apps and directory entries [SERVICE-617]
+        const runningApplications = await Promise.all((await fin.System.getAllApplications())
+            .filter(info => info.isRunning)
```

Comment:
Does `getAllApplication` ever return non-running apps?

Reply:
If I had to guess, probably not, but since the docs are silent on this, and since `isRunning` is specifically returned (and this isn't a shape that's used elsewhere - this is a different `ApplicationInfo` to the one returned by `app.getInfo()`), this feels like worthwhile safety

Reply:
Fair enough. I guess since it doesn't require another call into the core to check it's not much of a difference to keep it.